### DISCUSSION TITLE
Control codes stream into terminal after terminating the app

### DISCUSSION THREAD

**Discussion by AArnott** [tui-cs/Terminal.Gui#1392] [Q&A] [answered]
I almost always lose control of my terminal after using Ctrl-Q to terminate my TUI application to control codes like this. What's up with that? Am I doing something wrong?

![image](https://user-images.githubusercontent.com/3548/126878284-614c9000-5f43-41ce-847e-bfcccd0da96b.png)

**Comment by BDisp** [comment]
I still can not solve this. We have to type "`reset`" in the terminal to restore it.

**Reply by AArnott** [reply]
that doesn't stop it for me.

**Reply by tznind** [reply]
Does the `UICatalog` demo also trigger this behavior? Are you making sure to call `Application.Shutdown()`?

**Reply by BDisp** [reply]
> 
> 
> that doesn't stop it for me.

I suppose you're using `CursesDriver`. If this is the case, I forgot to mention that you have to type `reset`, before typing anything before. You will not see what you are typing. If you have pressed some key previously, press `Ctrl-C` and type `reset` again. Since you will not see what you are typing, be sure you are typing correctly.

I had already settled this problem before but the `CursesDriver` has become irresponsive, such as not returning the correct dimension of the terminal when resizing. I ended up reversing because the disadvantages were far higher than the advantages.

Surely that someone very familiar with programming with `ncurses` in `Linux` would solve this problem easily. I am just a mere curious without many knowledge in using the `ncurses` on `Linux`.

**Reply by BDisp** [reply]
> 
> 
> Does the `UICatalog` demo also trigger this behavior? Are you making sure to call `Application.Shutdown()`?

@tznind, as you use `Linux` a lot, you should also experience this behavior if you start typing after you have left an application with the `Terminal.Gui` library using the `CursesDriver`.

**Reply by BDisp** [reply]
I've been checking better and the `CursesDriver` does not leave that characters when you leave, possibly related to mouse movements. With the `CursesDriver` just you can not view the typing output on the terminal until we do not enter `reset`.
In this case, I think you should be using `NetDriver` on `Linux` that sometimes leaves a mouse movement trail. Not always this happens and so I still did not realize the reason for this.

**Comment by AArnott** [comment, marked as answer]
@tznind suggested in https://github.com/migueldeicaza/gui.cs/discussions/1392#discussioncomment-1046713 that my code call `Application.Shutdown()` before exiting. That seems to have resolved the issue. Thank you.
